src/test/java/commission/services/SaleServiceTest.java

```diff
@@ -0,0 +1,158 @@
+package commission.services;
+
+import commission.dao.SaleDaoImpl;
+import commission.entity.Sale;
+import commission.service.SaleServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class SaleServiceTest {
+
+    @Mock
+    private SaleDaoImpl saleDao;
+
+    @InjectMocks
+    private SaleServiceImpl saleService;
+
+
+    @BeforeEach
+    public void setUp(){
+        when(saleDao.findById(1)).thenReturn(Optional.empty());
+        when(saleDao.findById(2)).thenReturn(Optional.of(
+                new Sale(
+                        2,
+                        200,
+                        OffsetDateTime.parse("2024-07-22 13:36:48.401+0200",  DateTimeFormatter.ofPattern ( "yyyy-MM-dd HH:mm:ss.SSSX" )),
```

Comment:
Regarding your question on date / time creation: You don't need to specify the formatter if you use the `ISO` format. It is the default parser, this date format is `2024-07-22T13:36:48+02:00` then you can just use the `parse` method without explicitly give it a format.

Or you can use the `of(LocalDate, LocalTime, ZoneOffset)` method which would look something like this:
```
var time = OffsetDateTime.of(
  LocalDate.of(2024, Month.JULY, 22),
  LocalTime.of(13, 36),
  ZoneOffset.of("2") //Not sure about this one
);
```

Personally I like this, because it reads nicely and is easy to understand

For ease of reference you can also declare this time in at the class level and just use the variable in different tests so you don't need to always redefine it.

Reply:
When I leave the code without the formatter, I get the error. 
```
java.time.format.DateTimeParseException: Text '2024-07-22 13:36:48.401+0200' could not be parsed, unparsed text found at index <something>
```
That option does read better, but do I create a separate variable for it and then pass it like this? Or is there a different way that Im completely missing? 

```
        var timeOne = OffsetDateTime.of(
                LocalDate.of(2024, Month.JULY, 22),
                LocalTime.of(13, 36),
                ZoneOffset.of("2")
        );

        when(saleDao.findById(2)).thenReturn(Optional.of(new Sale(2, 200, timeOne, 12)));
```

PS: I hate time

Reply:
The parse exception is due to the format not being correct, you have to use the ISO standard format: `2024-07-22T13:36:48+02:00`. Note there is a `T` and no space between the date and time as well as a colon in the timezone.

Yup, you can set it to a variable like that, and you can even declare that variable at the class level so you can use it for multiple tests.

>PS: I hate time

Yup, it's really tricky to wrap your head around. Personally I can remember 2 or 3 timezone related SRs I caused 🤣